File: learn-dapr/learn-dapr/learn-dapr-api/Controllers/HelloController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using Dapr.AppCallback;

namespace learn_dapr_api.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class HelloController : ControllerBase
    {
        [HttpGet("ticks")] 
        public long GetTicks() => DateTime.Now.Ticks;
    }
}

File: learn-fsharp/fsharp-web/api-core/Controllers/HelloController.fs
namespace api_core.Controllers.HelloController

open System.Net.Http
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging

[<Route("api/hello")>]
[<ApiController>]
type HelloController(logger: ILogger<HelloController>, httpFactory: IHttpClientFactory) =
    inherit ControllerBase()

    [<HttpGet>]
    member _.Resp() =
        logger.LogError "This is a test!!!"

        task {
            logger.LogError "This is a test!!!"
            use client = httpFactory.CreateClient()
            let url = "https://www.baidu.com/"
            return! client.GetStringAsync(url)
        }
